standards/request-response.md

```diff
@@ -444,8 +444,32 @@ Access-Control-Allow-Methods: *
 
 <hr />
 
-```note
-**KEEP AN EYE ON IT**: The [Idempotency-Key Request header](https://tools.ietf.org/id/draft-idempotency-header-01.html) is in an experimental state but getting lots of attention as a pattern of making fault-tolerant resilient requests for traditionally non-idempotent methods like `POST`.
+#### [Idempotency-Key Request header](https://datatracker.ietf.org/doc/html/draft-idempotency-header-01)
+**Type**: Request
+
+**Support**: OPTIONAL
+
+**Description**:
+The HTTP Idempotency-Key request header field can be used to carry an "idempotency key" to make non-idempotent HTTP methods such as POST or PATCH fault-tolerant.
+This is an experimental header, as there is common usage of it, but no accepted spec for it.
+`Idempotency-Key` should typically be a V4 UUID as a string, or another random string with enough entropy to avoid collisions, and 
+should be no longer than 255 characters long.
+An API must in some way persist the Idempotency-Key in a stateful way to manage the idempotency.
+It is common to include the key as part of the primary key or as a secondary index for the resource.
+
+- The header value **MUST** follow this regex format: `[a-z0-9-]{20,255}`
+- An API **MUST** return a `400` when the header value is an invalid format.
+- On conflict, an API **MUST** return a `2xx` if the endpoint is idempotent, or a `409` if the endpoint is not idempotent.
+- An API **MAY** treat this as the final primary ID of the resource, or as some part of the ID.
```

Comment:
`[a-z0-9-]{20,255}` is included more for debate that thinking this is the best format.

I was on the fence of if this should be hard enforced as a UUID, but as I thought about it it seemed more flexible if it was UUID compatible rather than UUID forced.
There's advantages to it being a strict UUID, in that an API could assume it's a UUID and store it in UUID formats and use possibly more optimal storage for indexes.
There's disadvantage to forcing UUID in that you might want to accept a few kinds of things.

I think regardless of choice, I think it's useful to treat it like it could be a reference ID, and not allow weird non-url safe characters.  That does limit you to to essentially a hex format of some sort.  Either a UUID string hex format 
```
e9ac6f49-4963-4f20-a038-539c66b6f5db
```
or something like a sha-256 hex format
```
83d0437862c69e1b7d280b25ab68d8003f7b8af12a7ac3a47dd8c5a090c8c65c
```

Reply:
Ya that is interesting.

>There's disadvantage to forcing UUID in that there could be other styles of keys you might want to support, or use keys from various kinds of frameworks.
My thoughts immediately went to this, combined with your base64 example of how some usages of it might be represented.

If we don't have a strong opinion or clear direction on this, I think I'm more inclined to relax the format a bit.